triedb/pathdb/history_index.go

```diff
@@ -163,12 +163,15 @@ type indexWriter struct {
 	db       ethdb.KeyValueReader
 }
 
-// newIndexWriter constructs the index writer for the specified state.
-func newIndexWriter(db ethdb.KeyValueReader, state stateIdent) (*indexWriter, error) {
+// newIndexWriter constructs the index writer for the specified state. Additionally,
+// it takes an element ID and prunes all existing elements above that ID. It's
+// essential as the recovery mechanism after unclean shutdown during the history
+// indexing.
+func newIndexWriter(db ethdb.KeyValueReader, state stateIdent, limit uint64) (*indexWriter, error) {
```

Comment:
in the comment, `limit` is called an "element ID". The names should be consistent.